packages/react-components/lib/robots/robot-marker.tsx

```diff
@@ -0,0 +1,77 @@
+import { makeStyles } from '@material-ui/core';
+import * as RomiCore from '@osrf/romi-js-core-interfaces';
+import Debug from 'debug';
+import React from 'react';
+import { SvgText } from '..';
+import { DefaultMarker } from './default-marker';
+import { ImageMarker } from './image-marker';
+
+const debug = Debug('Robots:RobotMarker');
+
+const useStyles = makeStyles(() => ({
+  text: {
+    dominantBaseline: 'central',
+    textAnchor: 'middle',
+    fontSize: '0.18px',
+    fontWeight: 'bold',
+    fill: 'white',
+    /* 1 pixel black shadow to left, top, right and bottom */
+    textShadow: '-1px 0 black, 0 1px black, 1px 0 black, 0 -1px black',
+    pointerEvents: 'none',
+    userSelect: 'none',
+  },
+  container: {
+    pointerEvents: 'visible',
+    cursor: 'pointer',
+  },
+}));
+
+export interface RobotMarkerProps extends React.SVGAttributes<SVGGElement> {
+  robot: RomiCore.RobotState;
+  footprint: number;
+  fleetName: string;
+  iconPath?: string;
+  variant?: 'normal' | 'inConflict';
+}
+
+/**
+ * Contexts: ColorContext
+ */
```

Comment:
Are you putting this to know that this component is within the scope of the ColorContext context to use it in the DefaultMarker? In other words, it is required id you want to use the `DefaultMarker` component, right?

Reply:
Yeah, but it is not required, if the context is not given it will use the default `ColorManager`. The context is to allow consumers to override the default colors with their own implementation.